obelix_client/api.py

```diff
@@ -0,0 +1,247 @@
+                               file_format="PDF")
+
+    def log_page_view(self, user_info, recid, type="events.pageviews", file_format="view"):
+        """ Logs a page view """
+        # TODO: Maybe check if valid user
+        uid = user_info.get('uid', None)
+        ip = user_info.get('remote_ip', None)
+        uri = user_info.get('uri', None)
+
+        self.log_page_view_for_neo_feeder(uid, recid, ip, type, file_format)
+        self.log_page_view_for_analytics(uid, recid, ip, uri, type)
+
+    def log_page_view_for_neo_feeder(self, uid, recid, remote_ip, type, file_format):
+        """
+        Feed the Obelix NeoFeeder with new page views, used to construct the graph
+        :param uid:
+        :param recid:
+        :return: None
+        """
+        data = {
+            'item': recid,
+            'ip': remote_ip,
+            "type": type,
+            'user': uid,
+            'file_format': file_format,
+            "timestamp": time.time()
+        }
+        # goes to "logentries"
+        Obelix.signal_save_to_neo_feeder.send(self, data=data)
+
+    def log_page_view_for_analytics(self, uid, recid, ip, uri, type):
+        """ Mainly used to store statistics, may be removed in the future
+        :param uid:
+        :param recid:
+        :param ip:
+        :param uri:
+        :return:
+        """
+        storage_key = "{0}::{1}".format("last-search-result", uid)
+        last_search_info = self.cache.get(storage_key)
+
+        if not last_search_info:
+            return
+        # TODO: Check optimization
+        hit_number_global = 0
+        for collection_result in last_search_info['record_ids']:
+            if recid in collection_result:
+                hit_number_local = collection_result.index(recid)
+                hit_number_global += hit_number_local
+
+                timestamp = last_search_info['search_timestamp']
+                jrec = last_search_info['jrec']
+                rg = last_search_info['rg']
+                rm = last_search_info['rm']
+                cc = last_search_info['cc']
+
+                recommendations = self.recommendations.get(uid, {})
+                data = {'search_timestamp': timestamp,
+                        'recid': recid,
+                        'timestamp': time.time(),
+                        'uid': uid,
+                        'remote_ip': ip,
+                        'uri': uri,
+                        'jrec': jrec,
+                        'rg': rg,
+                        'rm': rm,
+                        'cc': cc,
+                        'hit_number_local': jrec + hit_number_local,
+                        'hit_number_global': jrec + hit_number_global,
+                        'recommendations': recommendations,
+                        'recid_in_recommendations': recid in recommendations,
+                        'type': type}
+                Obelix.signal_statistics_page_view.send(self, data=data)
```

Comment:
You should be able to use `self.signal_statistics_page_view.send` (easier subclassing)

--
PS: Please have a look to other occurrences too.